lib/manager/npm/extract/pnpm.ts

```diff
@@ -105,7 +105,7 @@ export async function detectPnpmWorkspaces(
       packageFilters !== null &&
       matchesAnyPattern(
         packageFile,
-        packageFilters.map((filter) => filter.replace(/\/?$/, '/package.json')) // TODO #12070 #12071
+        packageFilters.map((filter) => filter.replace(/\/?$/, '/package.json')) // TODO #12070 #12071 left now due to addition of / at front instead of end
```

Comment:
@rarkins  Should we use `joinUrlParts` here?

https://github.com/renovatebot/renovate/blob/87c5e8e0223a79264a1a25ddb73eec12595ed1cc/lib/util/url.ts#L4

Reply:
For now this code snippet:-
```js
 packageFilters.map((filter) => filter.replace(regEx(/\/?$/), '/package.json')) // TODO #12071
 ```
 works fine.That said changing it as per your suggestion will ease the debugging in future.